rdflib/plugins/sparql/sparql.py

```diff
@@ -264,11 +265,27 @@ def __init__(
         self.graph: Optional[Graph]
         self._dataset: Optional[ConjunctiveGraph]
         if isinstance(graph, ConjunctiveGraph):
-            self._dataset = graph
-            if rdflib.plugins.sparql.SPARQL_DEFAULT_GRAPH_UNION:
-                self.graph = self.dataset
+            if datasetClause:
+                self._dataset = ConjunctiveGraph()
+                self.graph = Graph()
+                for d in datasetClause:
+                    if d.default:
+                        self.graph += graph.get_context(d.default)
```

Comment:
Seems like its pretty costly to copy _every_ triple from each `FROM` graph into a new Graph. Could it be done by adding the existing graph to a new Dataset (or ConjunctiveGraph), and setting default_union on that store?

Reply:
If I understand well your suggestion, this was previously done in the load function:
`self.graph += self.dataset.get_context(source)  # type: ignore[operator]`
Exactly the same way.
So that's just something that I didn't change.
I will see if I can optimize that

Reply:
Thanks for the explanation. I see now this of course isn't something you added, but it was just to this codepath.
There is no need to include an optimization for that in this PR, because it could introduce an unintended change in behaviour to SPARQL processing.

I am planning on doing a large sprint of work on optimising the SPARQL processor after RDFLib v7.1.0 is released.
I suspect efficiency problems (exactly like this one) are behind the odd extremely slow execution users are seeing in issues like:
https://github.com/RDFLib/rdflib/issues/2147
and
https://github.com/RDFLib/rdflib/issues/1775
and 
https://github.com/RDFLib/rdflib/issues/787